caproto/threading/client.py

```diff
@@ -649,7 +660,11 @@ def command_loop(self):
                                     accepted_address, _ = search_results[name]
                                     new_address = ca.extract_address(command)
                                     if new_address != accepted_address:
-                                        self.log.warning(
+                                        pv_name_logger = logging.LoggerAdapter(search_logger, {
```

Comment:
I believe that we should only use `LoggerAdapter` which we want to emit a message with the _same_ extras multiple times. In this case, we are creating a `LoggerAdapter` instance only to then immediately drop it, so it would be more efficient to skip this step and pass the extras directly to `search_logger.warning("...", extra=...)`.